### ISSUE TITLE
DocblockTypeContradiction is not reported for function results (as opposed to variables)

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#1546]
https://psalm.dev/r/183262ea01
```php
<?php
/** @return false|string false when string is empty, first char of the parameter otherwise */
function firstChar(string $s) { return empty($s) ? false : $s[0]; }

$first = firstChar("sdf");
if (true === $first) { // Reported as DocblockTypeContradiction
}


if (true === firstChar("sdf")) { // Not reported
}
```
**Expected**: Both contradictions should be reported
**Actual**:
```
Psalm output (using commit a9b8952): 

ERROR: DocblockTypeContradiction - 6:5 - false|string does not contain true
```

**Comment by muglug** [comment]
There may be more where this came from, but thanks!

**Comment by muglug** [comment]
Yeah, I need to apply the same fix in more places
